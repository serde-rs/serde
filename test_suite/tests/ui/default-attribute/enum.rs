use serde_derive::Deserialize;

#[derive(Deserialize)]
#[serde(default)]
enum E {
    // No errors expected.
    T0(u8, u8),

    // No errors expected:
    // - If both fields are provided, both get value from data.
    // - If only one field is provided, the second gets default value.
    T1(u8, #[serde(default)] u8),

    // ERROR: The first field can get default value only if sequence is empty, but
    // that mean that all other fields cannot be deserialized without errors.
    T2(#[serde(default)] u8, u8, u8),

    // No errors expected:
    // - If both fields are provided, both get value from data.
    // - If only one field is provided, the second gets default value.
    // - If no fields are provided, both get default value.
    T3(#[serde(default)] u8, #[serde(default)] u8),

    S { f: u8 },
}

fn main() {}
