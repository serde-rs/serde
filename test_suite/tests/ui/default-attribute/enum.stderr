error: #[serde(default)] can only be used on structs
 --> tests/ui/default-attribute/enum.rs:4:9
  |
4 | #[serde(default)]
  |         ^^^^^^^

error: field must have #[serde(default)] because previous field 0 has #[serde(default)]
  --> tests/ui/default-attribute/enum.rs:16:30
   |
16 |     T2(#[serde(default)] u8, u8, u8),
   |                              ^^

error: field must have #[serde(default)] because previous field 0 has #[serde(default)]
  --> tests/ui/default-attribute/enum.rs:16:34
   |
16 |     T2(#[serde(default)] u8, u8, u8),
   |                                  ^^
